In [257]:
import numpy as np
import pandas as pd

Opening the traindata.txt file and input all datapoints into the array data and convert to numpy array (traindata)

In [258]:
train_data= open("traindata.txt", "r")
data=[]

for line in range(0,1000):
    data.append(train_data.readline().strip())

traindata=np.array(data)
#print(traindata)

Opening the trainlabel.txt file and input all labels into the array labels and convert to numpy array (trainlabel)

In [259]:
train_label=open("trainlabels.txt","r")
labels=[]

for line in range(0,1000):
    labels.append(int(train_label.readline().strip()))
trainlabel=np.array(labels)
#print(trainlabel)

split the feature in each datapoint and represent the training data as a 2D array of size (1000, 71)

In [279]:
data_train=[]
for i in range(0, 1000):
    #if trainlabel[i] == <insert the label you want>:
        #print("Datapoint", i+1)
        data= traindata[i].split(",")
        data = [float(i) for i in data]
        data_train.append(data)
        '''
        data=np.array(data)
        #print(len(data))
        #data= np.array(data).reshape(10,7)
        print(data)
        print("Label of data:", trainlabel[i], "\n")'''
#print(data_train)

### N.B: We can create a method to read the files and return an array :)

In [265]:
data_train=np.array(data_train).reshape(1000,71)
print(data_train)

[[-0.26  0.43 -0.3  ...  5.    1.    1.  ]
 [-0.15 -0.24 -0.68 ...  4.    1.    1.  ]
 [ 0.74 -1.09 -0.01 ...  5.    1.    0.  ]
 ...
 [ 0.17  0.01 -0.36 ...  2.    1.   -1.  ]
 [ 0.97  1.3  -0.   ...  9.   -1.   -1.  ]
 [ 0.01  0.16  0.35 ...  1.    1.   -1.  ]]


Now we use the gaussian naive bayes model from the sckitlearn library to fit our taining data to our training labels

In [280]:
from sklearn.naive_bayes import GaussianNB

Name our model GNBclf to define our gaussian NB classifier

In [281]:
GNBclf = GaussianNB()


We now get our test data and test label to check the accuracy of the model

In [282]:
data=[]
test_data=[]
testdata=open("traindata.txt", "r")
for line in range(0,2000):
    data.append(testdata.readline().strip())
    if line >= 1000:
        test_data.append(data[line])

testdata=np.array(test_data)
#print(testdata)

In [283]:
labels=[]
test_labels=[]
testlabel=open("trainlabels.txt", "r")
for line in range(0,2000):
    labels.append(int(testlabel.readline().strip()))
    if line >= 1000:
        test_labels.append(labels[line])
testlabel=np.array(test_labels)
#print(testlabel)

In [284]:
data_test=[]
for i in range(0, 1000):
    #if trainlabel[i] == <insert the label you want>:
        #print("Datapoint", i+1)
        data= testdata[i].split(",")
        data = [float(i) for i in data]
        data_test.append(data)
        '''
        data=np.array(data)
        #print(len(data))
        #data= np.array(data).reshape(10,7)
        print(data)
        print("Label of data:", trainlabel[i], "\n")'''
#print(data_test)

In [285]:
data_test=np.array(data_test).reshape(1000,71)
#print(data_test)

Fit the training data and the training labels, then test for the labels of our test data

In [273]:
predict_label=GNBclf.fit(data_train, trainlabel).predict(data_test)


Print out the predicted labels 

In [286]:
#predictlabel=np.array(predict_label).reshape(1000)
print(predictlabel)

[2 8 8 2 8 2 9 0 9 7 3 0 6 8 0 0 0 2 2 5 7 2 2 9 5 2 8 1 1 0 5 1 8 5 2 9 0
 5 7 5 1 2 8 0 2 3 2 3 5 5 8 9 4 5 0 2 8 6 3 2 3 9 3 7 4 1 0 7 9 3 8 5 0 7
 3 2 6 2 3 3 1 2 0 7 0 8 2 6 9 7 7 9 4 3 1 2 1 3 5 4 4 4 1 0 3 1 1 0 3 1 9
 6 9 9 9 6 9 9 1 2 5 7 0 6 5 0 9 0 2 1 3 1 3 2 2 9 9 5 9 0 7 0 0 2 3 7 5 7
 2 4 6 3 2 6 0 2 8 0 0 4 2 9 3 9 5 2 1 9 5 0 5 0 6 0 8 4 5 9 8 4 2 2 6 3 6
 3 7 5 2 7 2 9 2 8 2 6 5 3 3 2 1 5 8 9 6 9 5 8 0 1 6 2 7 7 9 7 0 9 6 9 5 8
 9 5 7 7 3 9 6 9 1 1 7 5 9 0 9 9 8 0 4 6 1 3 5 8 2 9 2 4 9 9 6 0 2 8 0 9 7
 1 8 0 2 1 5 8 5 5 8 3 3 9 0 9 8 8 3 1 5 1 5 9 8 3 1 1 2 2 2 1 7 7 8 1 7 5
 2 7 8 5 9 8 8 9 2 0 0 9 5 2 3 8 6 1 0 1 0 5 5 1 5 1 0 6 6 1 9 7 0 1 2 9 0
 9 3 9 5 6 7 1 6 3 0 8 9 5 7 9 6 5 9 1 3 9 4 6 5 5 8 8 7 7 8 4 9 2 7 4 2 1
 5 5 0 8 2 9 8 9 3 5 6 0 8 0 3 2 0 5 8 2 1 0 0 8 1 4 9 5 9 2 9 7 7 5 7 1 0
 3 7 5 9 5 7 8 7 6 0 9 7 3 5 1 8 0 7 6 0 1 5 5 5 1 0 1 9 1 5 0 2 4 0 0 1 0
 4 1 2 2 4 5 1 8 9 4 2 2 9 1 3 0 5 4 7 1 8 1 7 7 8 5 3 3 2 9 5 1 3 9 5 5 0
 8 3 1 1 7 3 1 4 0 3 0 3 

Function to test the accuracy of our model

In [287]:
def accuracy(real, predicted):
    accuracy=(predicted / real) * 100
    return accuracy

In [288]:
real_values=1000
correct=0
for i in range(0, 1000):
    if predictlabel[i] == testlabel[i]:
        correct+=1
print(correct)

164


In [289]:
print(accuracy(real_values,correct),"% accuracy")

16.400000000000002 % accuracy


...sigh...

Tried to test using the last 7 features, 15% accuracy was reached $$ $$
Tried to test using the first 64 features, still 16,4% was reached